### DISCUSSION TITLE
LS-GL error after last apt-get upgrade

### DISCUSSION THREAD

**Discussion by Enonir** [1000001101000/Debian_on_Buffalo#156] [Q&A]
Hello!
After my last week apt-get upgrade (some dbus changes in Debian written in console during this process) my LS-GL hangs after few minutes from boot.
I can login to this machine, work with bash, check logs, dmseg itd. Then everything is freezing, red down led blinks and my Buffalo sounds like beep beep beep beep. Maybe it is temperature alert.
How to debug this behavior?

**Comment by 1000001101000** [comment]
Good to hear from you!

This device has an onboard microcontroller that controls things like the LEDs and fan. One of its features is a watchdog timer which raises the alarm you are seeing if it doesn't receive the "BOOT_END" signal within a few minutes of startup. The code that I have been using to send that signal needs some improvement and can fail from time to time. Sometimes the board and/or microcontroller gets into a state where it won't accept signals anymore, presumable because the UART ports get misconfigured. 

You can likely see the errors occurring with this process via:
`systemctl status micon_boot.service` 

I believe if you shutdown the device then unplug it from power for a minute (so that the state of both the board and microcontroller are cleared) it should work properly when the device starts again. 

I plan to re-write this bit of code to make it more reliable someday though don't know when that will be.
